engine/servermaster/server.go

```diff
@@ -227,13 +226,11 @@ func (s *Server) Heartbeat(ctx context.Context, req *pb.HeartbeatRequest) (*pb.H
 
 	resp, err := s.executorManager.HandleHeartbeat(req)
 	if err == nil && resp.Err == nil {
-		s.members.RLock()
-		defer s.members.RUnlock()
-		addrs := make([]string, 0, len(s.members.m))
-		for _, member := range s.members.m {
-			addrs = append(addrs, member.AdvertiseAddr)
+		for _, nodeInfo := range s.discoveryKeeper.Snapshot() {
```

Comment:
Could `s.discoveryKeeper.Snapshot()` returns `nil` here, which will cause the client list of master client update to empty.

Reply:
it's unlikely that it returns nil here. For safety, we add an extra check here.